### PR TITLE
CDAP-17451 expose replicated tables in DeltaContext and add DeltaSource.initialize method

### PR DESCRIPTION
**Pull request by seanzhougoogle**
Some delta source need to know what are all the tables to be replicated so that it can initialize some resources based on the tables.

### CODE REVIEW

File: delta-api/src/main/java/io/cdap/delta/api/DeltaRuntimeContext.java [hunk 1]
```diff
@@ -79,4 +80,9 @@
    * @return the id of the delta pipeline
    */
   DeltaPipelineId getPipelineId();
+
+  /**
+   * @return all the Tables to be replicated
```
Comment (sagarkapare) [inline delta-api/src/main/java/io/cdap/delta/api/DeltaRuntimeContext.java]: nit: all the tables to be replicated by the current pipeline

File: delta-api/src/main/java/io/cdap/delta/api/DeltaRuntimeContext.java [hunk 2]
```diff
@@ -79,4 +80,9 @@
...
+   */
+  Set<SourceTable> getTables();
```
Comment (sagarkapare) [inline delta-api/src/main/java/io/cdap/delta/api/DeltaRuntimeContext.java]: nit: call it `getAllTables`?

File: delta-api/src/main/java/io/cdap/delta/api/DeltaSource.java [hunk 3]
```diff
@@ -35,6 +33,16 @@
...
   void configure(SourceConfigurer configurer);
 
+  /**
+   * Initialize the Delta Source stage. This is called after the application is deployed, whenever
+   * the program is initialized.
+   *
+   * @param context {@link DeltaSourceContext}
+   * @throws Exception if there is any error during initialization
+   */
+  default void initailize(DeltaSourceContext context) throws Exception {
```
Comment (sagarkapare) [inline delta-api/src/main/java/io/cdap/delta/api/DeltaSource.java]: nit: seems like type - `initialize`

**Review by sagarkapare — APPROVED** [review]
Few nits else LGTM.

File: delta-api/src/main/java/io/cdap/delta/api/DeltaRuntimeContext.java (hunk 1 above)
Comment (seanzhougoogle) [reply delta-api/src/main/java/io/cdap/delta/api/DeltaRuntimeContext.java]: done:)

File: delta-api/src/main/java/io/cdap/delta/api/DeltaRuntimeContext.java (hunk 2 above)
Comment (seanzhougoogle) [reply delta-api/src/main/java/io/cdap/delta/api/DeltaRuntimeContext.java]: done :)

File: delta-api/src/main/java/io/cdap/delta/api/DeltaSource.java (hunk 3 above)
Comment (seanzhougoogle) [reply delta-api/src/main/java/io/cdap/delta/api/DeltaSource.java]: done :)

**Pull request merged**